### DISCUSSION TITLE
"Renderer::renderItems() must be of the type array, null given" beim Aufruf des Embedded Videoplayers (ILIAS Version 7.26)

### DISCUSSION THREAD

**Discussion by pmanheller** [studer-raimann/ViMP#28] [General]
Hallo,
wir haben das ViMP-Plugin auf unserer ILIAS Instanz (v.7.26) im Einsatz. Wenn wir die Option "Embedded Videoplayer nutzen" in der Plugin-Konfiguration auswählen, erhalten wir beim Aufruf der Videos folgende Fehlermeldung:

TypeError thrown with message "Argument 1 passed to ILIAS\UI\Implementation\Component\Dropdown\Renderer::renderItems() must be of the type array, null given, called in /var/www/ilias/src/UI/Implementation/Component/Dropdown/Renderer.php on line 42"

Stacktrace:
#20 TypeError in /var/www/ilias/src/UI/Implementation/Component/Dropdown/Renderer.php:68
#19 ILIAS\UI\Implementation\Component\Dropdown\Renderer:renderItems in /var/www/ilias/src/UI/Implementation/Component/Dropdown/Renderer.php:42
#18 ILIAS\UI\Implementation\Component\Dropdown\Renderer:renderDropdown in /var/www/ilias/src/UI/Implementation/Component/Dropdown/Renderer.php:27
#17 ILIAS\UI\Implementation\Component\Dropdown\Renderer:render in /var/www/ilias/src/UI/Implementation/DefaultRenderer.php:46
#16 ILIAS\UI\Implementation\DefaultRenderer:render in /var/www/ilias/src/UI/Implementation/DefaultRenderer.php:65
#15 ILIAS\UI\Implementation\DefaultRenderer:renderAsync in /var/www/ilias/src/UI/Implementation/Render/DecoratedRenderer.php:87
#14 ILIAS\UI\Implementation\Render\DecoratedRenderer:renderAsync in /var/www/ilias/src/UI/Implementation/DefaultRenderer.php:62
#13 ILIAS\UI\Implementation\DefaultRenderer:renderAsync in /var/www/ilias/src/UI/Implementation/Render/DecoratedRenderer.php:87
#12 ILIAS\UI\Implementation\Render\DecoratedRenderer:renderAsync in /var/www/ilias/Customizing/global/plugins/Services/Repository/RepositoryObject/ViMP/src/UIComponents/Renderer/PlayerModalRenderer.php:130
#11 srag\Plugins\ViMP\UIComponents\Renderer\PlayerModalRenderer:renderComponents in /var/www/ilias/Customizing/global/plugins/Services/Repository/RepositoryObject/ViMP/src/UIComponents/Renderer/PlayerModalRenderer.php:86
#10 srag\Plugins\ViMP\UIComponents\Renderer\PlayerModalRenderer:render in /var/www/ilias/Customizing/global/plugins/Services/Repository/RepositoryObject/ViMP/classes/GUI/Abstract/class.xvmpGUI.php:344
#9 xvmpGUI:fillModalPlayer in /var/www/ilias/Customizing/global/plugins/Services/Repository/RepositoryObject/ViMP/classes/GUI/Abstract/class.xvmpGUI.php:212
#8 xvmpGUI:performCommand in /var/www/ilias/Customizing/global/plugins/Services/Repository/RepositoryObject/ViMP/classes/GUI/class.xvmpContentGUI.php:79
#7 xvmpContentGUI:performCommand in /var/www/ilias/Customizing/global/plugins/Services/Repository/RepositoryObject/ViMP/classes/GUI/Abstract/class.xvmpGUI.php:195
#6 xvmpGUI:executeCommand in /var/www/ilias/Services/UICore/classes/class.ilCtrl.php:217
#5 ilCtrl:forwardCommand in /var/www/ilias/Customizing/global/plugins/Services/Repository/RepositoryObject/ViMP/classes/class.ilObjViMPGUI.php:85
#4 ilObjViMPGUI:executeCommand in /var/www/ilias/Services/UICore/classes/class.ilCtrl.php:217
#3 ilCtrl:forwardCommand in /var/www/ilias/Services/Repository/classes/class.ilObjPluginDispatchGUI.php:45
#2 ilObjPluginDispatchGUI:executeCommand in /var/www/ilias/Services/UICore/classes/class.ilCtrl.php:217
#1 ilCtrl:forwardCommand in /var/www/ilias/Services/UICore/classes/class.ilCtrl.php:178
#0 ilCtrl:callBaseClass in /var/www/ilias/ilias.php:23

Könnt ihr mir hier weiterhelfen?

Viele Grüße
Peter

**Comment by iljalukin** [comment]
Hallo,
mir fällt spontan kein Grund ein, warum der eingebettete Player verwendet werden sollte, aber das Problem kann wie folgt umgangen werden:

```diff
--- a/classes/GUI/Abstract/class.xvmpGUI.php
+++ b/classes/GUI/Abstract/class.xvmpGUI.php
@@ -131,6 +131,7 @@ abstract class xvmpGUI {
         $popover = $this->dic->ui()->factory()->popover()->standard(
             $this->dic->ui()->factory()->legacy($this->pl->txt('popover_link_copied')));
 
+        $items = [];
         if (!xvmpConf::getConfig(xvmpConf::F_EMBED_PLAYER)) {
             $items[] = $this->dic->ui()->factory()->button()->shy($this->pl->txt('btn_copy_link_w_time'),
                 '')->withOnClick($popover->getShowSignal())->withOnLoadCode(function ($id) use ($link_tpl) {
```

**Reply by pmanheller** [reply]
Hallo,
mit dem internen Player werden bei uns die Videos nicht abgespielt - der Player verharrt in der Ladeanimation. Sowohl client- als auch serverseitig wird kein Fehler geworfen. Clientseitig wird die smil-Datei ständig neu geladen. Deshalb der Grund den eingebetteten Player zu verwenden und der Hoffnung dem Problem kurzfristig zu entgehen.
Danke für den Fix - Videos werden nun abgespielt!
